### DISCUSSION TITLE
Keywords can't contain special characters?

### DISCUSSION THREAD

**Discussion by ghost** [SpartanJ/ecode#216] [General]
Keywords beginning with `#` are preprocessors in FreeBASIC (like in C). QB64 keywords have suffixes. For example, `input$`. I found that ecode ignores all keywords with special characters like that in their names.

**Comment by SpartanJ** [comment]
They can, but "symbol" is usually identified with a word pattern, you can add any other pattern to identify them, it happens that usually keywords are words. Check any language definition and you will find that always the last pattern is something like this:
```
                          { { "[%a_][%w_]*" }, "symbol" },
```

This translates to "for any word try to look-up any symbol declared", being the symbol = keywords.
So you can add any pattern to do the symbol look-up if needed.
